core/domain/acl_decorators_test.py

```diff
@@ -0,0 +1,121 @@
+# limitations under the License.
+
+from core.domain import acl_decorators
+from core.controllers import base
+from core.domain import rights_manager
+from core.tests import test_utils
+import feconf
+import webtest
+import webapp2
+
+
+class PlayExplorationDecoratorTest(test_utils.GenericTestBase):
+    """Tests for play exploration decorator."""
+    published_exp_id = 'exp_id_1'
+    private_exp_id = 'exp_id_2'
+
+    class MockHandler(base.BaseHandler):
+        @acl_decorators.can_play_exploration
+        def get(self, exploration_id):
+            return self.render_json({'exploration_id': exploration_id})
+
+    def setUp(self):
+        super(PlayExplorationDecoratorTest, self).setUp()
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+        self.set_admins([self.ADMIN_USERNAME])
+        self.testapp = webtest.TestApp(webapp2.WSGIApplication(
+            [webapp2.Route('/mock/<exploration_id>', self.MockHandler)],
+            debug=feconf.DEBUG,
+        ))
+        self.save_new_valid_exploration(
+            self.published_exp_id, self.owner_id)
+        self.save_new_valid_exploration(
+            self.private_exp_id, self.owner_id)
+        rights_manager.publish_exploration(
+            self.owner_id, self.published_exp_id)
+
+    def test_guest_can_access_published_exploration(self):
+        response = self.get_json('/mock/%s' % self.published_exp_id)
+        self.assertEqual(response['exploration_id'], self.published_exp_id)
+
+    def test_admin_can_access_private_exploration(self):
+        self.login(self.ADMIN_EMAIL)
+        response = self.get_json('/mock/%s' % self.private_exp_id)
+        self.assertEqual(response['exploration_id'], self.private_exp_id)
+        self.logout()
+
+    def test_owner_can_access_private_exploration(self):
+        self.login(self.OWNER_EMAIL)
+        response = self.get_json('/mock/%s' % self.private_exp_id)
+        self.assertEqual(response['exploration_id'], self.private_exp_id)
+        self.logout()
+
+
+class PlayCollectionDecoratorTest(test_utils.GenericTestBase):
+    """Tests for play collection decorator."""
+    published_exp_id = 'exp_id_1'
+    private_exp_id = 'exp_id_2'
+    published_col_id = 'col_id_1'
+    private_col_id = 'col_id_2'
+
+    class MockHandler(base.BaseHandler):
+        @acl_decorators.can_play_collection
+        def get(self, collection_id):
+            return self.render_json({'collection_id': collection_id})
+
+    def setUp(self):
+        super(PlayCollectionDecoratorTest, self).setUp()
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+        self.set_admins([self.ADMIN_USERNAME])
+        self.testapp = webtest.TestApp(webapp2.WSGIApplication(
```

Comment:
Your test app does not have an /admin or signup handler, so probably those requests are hitting endpoints that don't exist.